src/IpcEventDispatcher.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "IpcEventDispatcher.h"
+#include "IpcChannel.h"
+#include "PointViewerMainWindow.h"
+
+
+IpcEventDispatcher::IpcEventDispatcher(QObject* parent, PointViewerMainWindow* getPayload, 
+                                       QByteArray hookSpec, QByteArray hookPayload, int whichShortCut)
+    : QObject(parent),
```

Comment:
This is an interesting design, making `IpcEventDispatcher` a child of the channel.  It's not what I had in mind, but from what I can see so far it seems like a pretty good idea.

In this design, the event dispatcher is more of an event formatting class, rather than explicitly dispatching the event to the underlying channel.  Perhaps `IpcEventFormatter` would make more sense?